datafusion/src/physical_plan/hash_join.rs

```diff
@@ -954,6 +1028,26 @@ pub fn create_hashes<'a>(
                     multi_col
                 );
             }
+            DataType::Date32 => {
```

Comment:
Did you have to add these match arms because tests failed without doing so?

Reply:
Yes